pkg/agent/core/ngt/service/ngt.go

```diff
@@ -96,110 +117,162 @@ const (
 	kvsFileName = "ngt-meta.kvsdb"
 )
 
-func New(cfg *config.NGT) (nn NGT, err error) {
+func New(cfg *config.NGT, opts ...Option) (nn NGT, err error) {
 	n := new(ngt)
-	n.inMem = cfg.EnableInMemoryMode
-	cfg.IndexPath = strings.TrimSuffix(cfg.IndexPath, "/")
-	opts := []core.Option{
+
+	for _, opt := range append(defaultOpts, opts...) {
+		if err := opt(n); err != nil {
+			return nil, errors.ErrOptionFailed(err, reflect.ValueOf(opt))
+		}
+	}
+
+	n.kvs = kvs.New()
+
+	err = n.initNGT(
 		core.WithInMemoryMode(n.inMem),
-		core.WithIndexPath(cfg.IndexPath),
+		core.WithIndexPath(n.path),
 		core.WithDimension(cfg.Dimension),
 		core.WithDistanceTypeByString(cfg.DistanceType),
 		core.WithObjectTypeByString(cfg.ObjectType),
 		core.WithBulkInsertChunkSize(cfg.BulkInsertChunkSize),
 		core.WithCreationEdgeSize(cfg.CreationEdgeSize),
 		core.WithSearchEdgeSize(cfg.SearchEdgeSize),
+		core.WithDefaultPoolSize(cfg.DefaultPoolSize),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if n.dur == 0 || n.alen == 0 {
+		n.dcd = true
+	}
+	if n.ivc == nil {
+		n.ivc = new(vcaches)
+	}
+	if n.dvc == nil {
+		n.dvc = new(vcaches)
 	}
 
-	if !n.inMem && len(cfg.IndexPath) != 0 {
-		n.path = cfg.IndexPath
+	if in, ok := n.indexing.Load().(bool); !ok || in {
```

Comment:
when we call `New` method, indexing valueable always contains nil.
so I think the following is better 🤔 

 ```suggestion
	n.indexing.Store(false)
```

Reply:
Thanks! i'll fix it.